kroxylicious/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyFrontendHandler.java

```diff
@@ -140,7 +140,7 @@ public void channelRead(final ChannelHandlerContext ctx, Object msg) {
     }
 
     public void outboundWritabilityChanged(ChannelHandlerContext outboundCtx) {
-        assert this.outboundCtx == outboundCtx;
+        LOGGER.warn("Outbound Channel write ability change notification for inappropriate context. {} received notification for {}", this.outboundCtx, outboundCtx);
```

Comment:
Should this be guarded by an `if (this.outboundCtx != outboundCtx)`? 
Should we use the single word 'writability'

Reply:
Yes, this certainly should be guarded and I'll go have a word with the IntelliJ spell checker.